src/services/db/CommitServiceGithub.ts

```diff
@@ -0,0 +1,77 @@
+import { AxiosCacheInstance } from "axios-cache-interceptor"
+
+import UserWithSiteSessionData from "@root/classes/UserWithSiteSessionData"
+import isFileAsset from "@root/utils/commit-utils"
+import { isReduceBuildTimesWhitelistedRepo } from "@root/utils/growthbook-utils"
+import GitHubService, { STAGING_LITE_BRANCH } from "@services/db/GitHubService"
+
+import GitFileSystemService from "./GitFileSystemService"
+
+/**
+ * Responsibilities of this class
+ * 1. Creates all commits to staging
+ * 2. Creates non-asset related commits to staging-lite
+ */
+export default class CommitServiceGitHub extends GitHubService {
+  private readonly STAGING_BRANCH = "staging"
+
+  private readonly STAGING_LITE_BRANCH = "staging-lite"
+
+  private readonly gitFileSystemService: GitFileSystemService
+
+  constructor(
+    gitFileSystemService: GitFileSystemService,
+    axiosInstance: AxiosCacheInstance
+  ) {
+    super({ axiosInstance })
+    this.gitFileSystemService = gitFileSystemService
+  }
+
+  async create(
+    sessionData: UserWithSiteSessionData,
+    {
+      content,
+      fileName,
+      directoryName,
+      isMedia = false,
+    }: {
+      content: string
+      fileName: string
+      directoryName: string
+      isMedia?: boolean
+    }
+  ): Promise<{
+    sha: string
+  }> {
+    const createPromises = [
+      super.create(sessionData, {
+        content,
+        fileName,
+        directoryName,
+        isMedia,
+      }),
+    ]
+    const shouldStagingLiteUpdate =
+      !isFileAsset(directoryName) &&
+      isReduceBuildTimesWhitelistedRepo(sessionData.growthbook)
+
+    // todo check if directory name is works here
+    if (shouldStagingLiteUpdate) {
+      createPromises.push(
+        super.create(sessionData, {
+          content,
+          directoryName,
+          fileName,
+          isMedia,
+          branchName: STAGING_LITE_BRANCH,
+        })
+      )
+    }
+
+    // We still await and throw error if the commit to staging-lite fails,
+    // but the caller only gets the `resultToStaging` commit returned
+    const [resultToStaging, _] = await Promise.all(createPromises)
```

Comment:
just sanity check - promise.all will reject upon first promise failure. any possibility staging and staging-lite will diverge? I think probably not as we will rollback, but just re-checking

Reply:
No there is a case that might occur here. currently, the promise fails if ANY one of the promises fail, and eng would have to intervene to keep staging + staging lite the same (operation will still result in an error for the user). we could have a rollback handler here to keep staging + staging-lite in sync, but I do feel that it does not have to go in by today. 
I can add in a jira ticket and priortise accordingly if you feel that this is an important, thoughts?

Reply:
But create failed operation does rollback right? src/services/db/GitFileSystemService.ts

but in any case, to answer the qn, addressing this in a subsequent PR is fine

Reply:
does not for github failures, agreed with chin here. addessing it post release